domain/src/Commands/DomainCommands.php

```diff
@@ -0,0 +1,1288 @@
+          ['!url' => $domain->getPath()]));
+      }
+    }
+  }
+
+  /**
+   * Sets the default domain. If run from a subfolder, specify the --uri.
+   *
+   * @param $domain_id
+   *   The numeric id or hostname of the domain to make default.
+    * @param array $options
+   *    An associative array of options whose values come from cli, aliases,
+   *    config, etc.
+   * @option validate
+   *   Force a check of the URL response before allowing registration.
+   * @usage drush domain-default www.example.com
+   * @usage drush domain-default example_org
+   * @usage drush domain-default www.example.org --validate=1
+   *
+   * @command domain:default
+   * @aliases domain-default
+   *
+   * @return string
+   *   Ihe machine name of the default domain.
+   *
+   * @throws \Drupal\domain\Commands\DomainCommandException
+   */
+  public function defaultDomain($domain_id, array $options = ['validate' => null]) {
+    $domain_storage = $this->getStorage();
+    // Resolve the domain.
+    if (!empty($domain_id) && $domain = $this->getDomainFromArgument($domain_id)) {
+      $validate = ($options['validate']) ? 1 : 0;
+      $domain->addProperty('validate_url', $validate);
+      if ($error = $this->checkHTTPResponse($domain)) {
+        throw new DomainCommandException(dt('Unable to verify domain !dom: !error',
+          ['!dom' => $domain->getHostname(), '!error' => $error]));
+      }
+      else {
+        $domain->saveDefault();
+      }
+    }
+
+    // Now, ask for the current default, so we know if it worked.
+    $domain = $domain_storage->loadDefaultDomain();
+    if ($domain->status()) {
+      $this->logger()->info(dt('!domain set to primary domain.',
+        ['!domain' => $domain->getHostname()]));
+    }
+    else {
+      $this->logger()->warning(dt('!domain set to primary domain, but is also inactive.',
+        ['!domain' => $domain->getHostname()]));
+    }
+    return $domain->id();
+  }
+
+  /**
+   * Deactivates the domain.
+   *
+   * @param $domain_id
+   *   The numeric id or hostname of the domain to disable.
+   * @usage drush domain-disable example.com
+   * @usage drush domain-disable 1
+   *
+   * @command domain:disable
+   * @aliases domain-disable
+   *
+   * @return string
+   *   'disabled' if the domain is now disabled.
+   *
+   * @throws \Drupal\domain\Commands\DomainCommandException
+   */
+  public function disable($domain_id) {
+    // Resolve the domain.
+    if ($domain = $this->getDomainFromArgument($domain_id)) {
+      if ($domain->status()) {
+        $domain->disable();
+        $this->logger()->info(dt('!domain has been disabled.',
+          ['!domain' => $domain->getHostname()]));
+      }
+      else {
+        $this->logger()->info(dt('!domain is already disabled.',
+          ['!domain' => $domain->getHostname()]));
+      }
+      if ($domain->status()) {
+        return 'enabled';
+      }
+      else {
+        return 'disabled';
+      }
+    }
+    return 'unknown domain';
```

Comment:
Needs `dt()`? Needs capitalization?

Reply:
I'm not sure. Drush now more than before encourages subcommands to return data so it can be processed by the drush output system, so here I am returning a string which will be printed, possibly as json, ...

If this is the intent, then dt() is not really wanted, because we want predictable returned value. However for human invocations then dt() would be appropriate.

Reply:
This is a drush change that I'm not aware of, so this seems fine. Any idea how this is reported? That would answer the capitalization question.